metals/src/main/scala/scala/meta/internal/builds/ShellRunner.scala

```diff
@@ -130,3 +133,34 @@ class ShellRunner(
   }
 
 }
+
+object ShellRunner {
+
+  def runSync(
+      args: List[String],
+      directory: AbsolutePath,
+      redirectErrorOutput: Boolean,
+      additionalEnv: Map[String, String] = Map.empty,
+      processOut: String => Unit = scribe.info(_),
+      processErr: String => Unit = scribe.error(_),
+      propagateError: Boolean = false,
+      maybeJavaHome: Option[String]
+  ): Option[String] = {
+    val env = additionalEnv ++ maybeJavaHome.map("JAVA_HOME" -> _).toMap
+    val ps = SystemProcess.run(
+      args,
+      directory,
+      redirectErrorOutput,
+      env,
+      Some(processOut),
+      Some(processErr),
+      propagateError
+    )
+
+    val scanner = new Scanner(ps.inputStream).useDelimiter("\\A")
+    if (scanner.hasNext)
+      Some(scanner.next)
+    else None
```

Comment:
```suggestion
   Await.result(ps.complete, ...)
```
I don't think it's working yet, is it?

Reply:
it is.

Reply:
added automated test in JdkVersionSuite.

Reply:
the test is flaky. So I would return to the previous logic when it was returning the Future and awaiting its result.

Reply:
I just wonder what invocation is blocking in the function

Reply:
perhaps it sometimes reads the inputStream before the process completes and hence, there is nothing to be read from it.